### PR TITLE
Enforce that no-pattern hints do not have an associated global.

### PR DESCRIPTION
**Pull request by ppedrot**
Trivial cleanup.

### CODE REVIEW

File: tactics/hints.ml [hunk 1]
```diff
@@ -675,10 +674,10 @@ struct
     let idv = id, { v with db = db.hintdb_name } in
       match gr with
       | None ->
-          let is_present (_, (_, v')) = KerName.equal v.code.uid v'.code.uid in
+          let is_present (_, v') = KerName.equal v.code.uid v'.code.uid in
           if not (List.exists is_present db.hintdb_nopat) then
             (* FIXME *)
```
Comment (SkySkimmer) [inline tactics/hints.ml:679]: Is this FIXME about what this PR does or something else?

Comment (ppedrot) [reply tactics/hints.ml:679]: I think that it's about the O(n) check in the list, so unrelated.

**Review by SkySkimmer — APPROVED** [review]
seems ok

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**